go/apps/ctrl/services/build/backend/depot/create_build.go

```diff
@@ -367,3 +293,73 @@ func (s *Depot) getOrCreateDepotProject(ctx context.Context, unkeyProjectID stri
 
 	return depotProjectID, nil
 }
+
+func (s *Depot) processBuildStatus(
+	statusCh <-chan *client.SolveStatus,
+	workspaceID, projectID, deploymentID string,
+) {
+	aggregator := NewBuildStepAggregator()
+	for status := range statusCh {
+		aggregator.ProcessStatus(
+			status,
+			workspaceID,
+			projectID,
+			deploymentID,
+			s.clickhouse.BufferBuildStep,
+			s.clickhouse.BufferBuildStepLog,
+		)
+	}
+}
+
+type BuildStepAggregator struct {
```

Comment:
this feels unnecessary

we're already spawning a goroutine to handle logs from this build, why do we encapsulate it in a `BuildStepAggregator` concept if we can just do everything in the `processBuildStatus` method?

Reply:
I mean it just adds complexity for now reason

Reply:
wdyt about flattening it like this?


```go

func (s *Depot) processBuildStatus(
	statusCh <-chan *client.SolveStatus,
	workspaceID, projectID, deploymentID string,
) {
	
	completed      :=  map[digest.Digest]bool{}
	verticesWithLogs := map[digest.Digest]bool{}
	
	
	for status := range statusCh {
   	....
	}
}
```